extensions/wikia/TemplateClassification/TemplateClassificationController.class.php

```diff
@@ -49,4 +54,76 @@ public function getTemplateClassificationEditForm() {
 
 		$this->setVal( 'templateTypes', $templateTypes );
 	}
+
+	/**
+	 * Classify all templates with given category
+	 *
+	 * @requestParam string category category name
+	 * @requestParam string type template type name
+	 *
+	 * @throws ApiException
+	 * @throws \BadRequestApiException
+	 * @throws \InvalidParameterApiException
+	 * @throws \MWException
+	 * @throws \UnauthorizedException
+	 */
+	public function classifyTemplateByCategory() {
+		global $wgCityId;
+
+		$errors = [];
+		$category = $this->request->getVal( 'category', null );
+		$templateType = $this->request->getVal( 'type', null );
+		$userId = $this->wg->User->getId();
+
+		$this->validateRequestForBulkEdit( $this->wg->User, $this->request, $category, $templateType );
+
+		$templates = ( new Helper() )->getTemplatesByCategory( $category );
+		$utcs = new \UserTemplateClassificationService();
+
+		foreach ( $templates as $templateId => $templateTitle ) {
+			try {
+				$utcs->classifyTemplate($wgCityId, $templateId, $templateType, 'user', $userId );
+			} catch( ApiException $e ) {
+				$errors[] = \Title::newFromText( $templateTitle )->getText();
+			}
+		}
+
+		if ( !empty( $errors ) ) {
+			$this->prepareBulkActionError( $errors, $templates );
+		}
+	}
+
+	private function validateRequestForBulkEdit( \User $user, \WikiaRequest $request, $category, $templateType ) {
+		if ( !$request->wasPosted() ) {
+			throw new \BadRequestApiException();
+		}
+
+		if ( !$user->isAllowed( 'template-bulk-classification' )
+			|| !$user->matchEditToken( $request->getVal( 'editToken' ) )
+		) {
+			throw new \UnauthorizedException();
+		}
+
+		if ( empty( $category ) ) {
+			throw new \InvalidParameterApiException( 'category' );
+		}
+
+		if ( empty( $templateType ) ) {
+			throw new \InvalidParameterApiException( 'template type' );
+		}
+
+		return true;
+	}
+
+	private function prepareBulkActionError( Array $errors, Array $templates ) {
+		if ( count( $errors ) === count( $templates ) ) {
+			$errorMessage = wfMessage( 'template-classification-edit-modal-error' )->escaped();
+		} else {
+			$errorMessage = wfMessage(
+				'template-classification-edit-modal-bulk-error',
+				implode( ',', $errors )
```

Comment:
Consider using `Language::listToText` as it localises the separator and adds a localised "and" between the last two items.